accepted/future-releases/language-versioning/language-versioning.md

```diff
@@ -68,11 +67,17 @@ If a `pubspec.yaml` does not contain a minimum SDK version, the default language
 
-The syntax for defaulting to a lower language level is a comment line of the form `@dart=2.0` (allowing for whitespace except inside the `@dart` or the version) in the initial part of the file, prior to any language declarations. So, it can be preceded by a `#!` line or by other comments, but not by a `library` declaration or any other declaration. Part files cannot be marked, they are always treated the same way as the library they belong to. All tools processing source files must be able to recognize this marker and process the library accordingly. 
+The syntax for defaulting to a lower language level is line consisting only of a single-line comment of the form `// @dart = 2.0` (exactly two slashes, the string `@dart`, a `=`, and a numeral which is two decimal numbers separated by `.`) in the initial part of the file, prior to any language declarations. It can be preceded by a `#!` line or by other comments, but not by a `library` declaration or any other declaration. Whitespace (space and tabs) are allowed everywhere except between the slashes, inside the `@dart` string and inside the version numeral. The comment must be an actual single-line comment, and not a string of that form embedded in a block comment. Block comments traditionally start lines with a `*` character, but that is purely convention, so it would be possible to have a line containing just  `//@dart=2.3` inside a block comment. That will not count as a language version override marker.
 
-Examples:
+This does mean that if a later language version change the *comment syntax* in a non-backwards compatible way, we may have to add further restrictions at that point. That is unlikely to happen. 
+
+Part files may be marked, but are not required to be. They must always be treated the same way as the library they belong to, so it is a compile-time error if a part file has a different language level override than its library. Tools that work on individual part files, like the formatter, can use a marker in the part file, but if there isn't one, the tool should make an effort to find the library that the file belongs to, and use the version of that library. If the part uses a URI-based `part of` declaration, the library file should be accessible. If it uses a name-based `part of` declaration, the tool might not be able to find the library from the part file alone, and it might be prudent to add a language level override to the part file as well.
```

Comment:
This seems like the worst of both worlds.  Do we want tools to support this or not?  If all we can tell users is "tools may or may not support this", then they de facto have to write it, and yet at the same time we're asking tools to put effort into trying to support it sometimes?

Reply:
We can go with "part files must have the mark" too. I'm warming to the idea. After all, we don't mind making it uncomfortable to *not* migrate.

Let's do that.

Reply:
SGTM, thanks!